addons/hr_contract/models/resource.py

```diff
@@ -28,17 +31,35 @@ def transfer_leaves_to(self, other_calendar, resources=None, from_date=None):
             'calendar_id': other_calendar.id,
         })
 
+    @api.depends('contract_ids', 'contract_ids.state')
     def _compute_contracts_count(self):
-        count_data = self.env['hr.contract']._read_group(
-            [('resource_calendar_id', 'in', self.ids)],
-            ['resource_calendar_id'],
-            ['__count'])
-        mapped_counts = {resource_calendar.id: count for resource_calendar, count in count_data}
+        """ Compute total and running contract counts in a single query. """
+        contracts_data = self.env['hr.contract']._read_group(
+            [('resource_calendar_id', 'in', self.ids), ('company_id', '=', self.env.company.id)],
```

Comment:
```suggestion
            [('resource_calendar_id', 'in', self.ids), ('company_id', 'in', self.env.companies.ids)],
```
it's a bit sad to not have this compute 
![image](https://github.com/user-attachments/assets/f95abd4d-3d01-4c5e-9301-f516c4c868f5)
![image](https://github.com/user-attachments/assets/bf89551b-5b9b-477b-ad99-ca5560418be3)